Makefile

```diff
@@ -1,13 +1,24 @@
-package: 
+.PHONY: clippy dev format lint package run scratch test ts_test test_application test_rust test_rehash
+
+ifeq ($(OSTYPE),darwin)
+PACKAGE := application/release/mac/WereSoCool.app
+else
+PACKAGE := application/release/linux-unpacked/weresocool
+endif
+
+$(PACKAGE): application/node_modules
 	cd application && yarn package
 
-run:
-	./application/release/linux-unpacked/weresocool
+package: $(PACKAGE)
 
-run_osx:
-	open ./application/release/mac/WereSoCool.app
+run: $(PACKAGE)
+ifeq ($(OSTYPE),darwin)
+	open $(PACKAGE)
+else
+	$(PACKAGE)
+endif
 
-dev: 
+dev: application/node_modules
 	(cd application && yarn build-backend && yarn dev)
```

Comment:
Adding `application/node_modules` as a dependency to a target will cause make to run the `application/node_modules` target first if it needs updating.